packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_app_and_identity.yml

```
@@ -1,468 +1,559 @@
-      if: ctx.json?.properties?.RawEventData != null && ctx.json.properties.RawEventData instanceof String && ctx.json.properties.RawEventData != ''
-      on_failure:
-        - append:
-            field: error.message
-            value: '{{{_ingest.on_failure_message}}}'
   - rename:
       field: json.properties.RawEventData
       target_field: m365_defender.event.raw_event_data
+      tag: rename_json_properties_RawEventData
       ignore_missing: true
   - rename:
       field: json.properties.UserAgentTags
       target_field: m365_defender.event.user_agent_tags
+      tag: rename_json_properties_UserAgentTags
+      ignore_missing: true
+  
+  # Destination mapping
+  - set:
+      field: destination.domain
+      copy_from: m365_defender.event.target.device_name
+      tag: set_destination_domain
+      ignore_empty_value: true
+  - set:
+      field: destination.ip
+      copy_from: m365_defender.event.destination.ip_address
+      tag: set_destination_ip
+      ignore_empty_value: true
+  - set:
+      field: destination.port
+      copy_from: m365_defender.event.destination.port
+      tag: set_destination_port
+      ignore_empty_value: true
+
+  # Source mapping
+  - set:
+      field: source.geo.city_name
+      copy_from: m365_defender.event.city
+      tag: set_source_geo_city_name
+      ignore_empty_value: true
+  - set:
+      field: source.geo.country_iso_code
+      copy_from: m365_defender.event.country_code
+      tag: set_source_geo_country_iso_code
+      ignore_empty_value: true
+
+  # Host mapping
+  - set:
+      field: host.name
+      copy_from: m365_defender.event.device.name
+      tag: set_host_name
+      ignore_empty_value: true
+  - lowercase:
+      field: host.name
+      tag: lowercase_host_name
+      if: ctx.host?.name != null
+  - set:
+      field: host.ip
+      copy_from: m365_defender.event.ip_address
+      tag: set_host_ip
+      ignore_empty_value: true
+  - set:
+      field: host.os.name
+      copy_from: m365_defender.event.os.platform
+      tag: set_host_os_name
+      ignore_empty_value: true
+  - set:
+      field: host.type
+      copy_from: m365_defender.event.device.type
+      tag: set_host_type
+      ignore_empty_value: true
+
+  # Event Outcome/Action mapping
+  - set:
+      field: event.action
+      copy_from: m365_defender.event.action.type
+      tag: set_event_action
+      ignore_empty_value: true
+  - lowercase:
+      field: event.action
+      tag: lowercase_event_action
+      ignore_missing: true
+  - gsub:
+      field: event.action
+      pattern: ' '
+      replacement: '-'
+      tag: gsub_event_action
+      ignore_missing: true
+  - set:
+      field: event.outcome
+      value: 'success'
+      tag: set_event_outcome
+      if: (ctx.m365_defender?.event?.failure_reason == null || ctx.m365_defender.event.failure_reason == '') && ctx.m365_defender?.event?.category != null && ctx.m365_defender?.event?.category?.toLowerCase() == 'advancedhunting-identitylogonevents'
```

```suggestion
      if: (ctx.m365_defender?.event?.failure_reason == null || ctx.m365_defender.event.failure_reason == '') && ctx.m365_defender?.event?.category != null && ctx.m365_defender.event.category.toLowerCase() == 'advancedhunting-identitylogonevents'
```
and below, though invariant comment stands.